agent/agent_test.go

```diff
@@ -424,22 +422,25 @@ func TestAgent_Session_TTY_MOTD_Update(t *testing.T) {
 		},
 	}
 
-	const updateInterval = 100 * time.Millisecond
-
 	ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
 	defer cancel()
 	//nolint:dogsled // Allow the blank identifiers.
 	conn, client, _, _, _ := setupAgent(t, &client{}, 0)
 	for _, test := range tests {
 		test := test
-		// Set new banner func and wait for the agent to call it to update the
-		// banner.
+
+		ready := make(chan struct{}, 1)
 		client.mu.Lock()
 		client.getServiceBanner = func() (codersdk.ServiceBannerConfig, error) {
+			select {
+			case ready <- struct{}{}:
+			default:
+			}
 			return test.banner, nil
 		}
 		client.mu.Unlock()
-		time.Sleep(updateInterval)
+		<-ready
+		<-ready // Wait for two updates to ensure the value has propagated.
```

Comment:
Review: This is the race fix, if we only wait for one `ready`, the service banner may not yet be stored in the agents atomic pointer.

Reply:
Ahh good find!  Thank you for fixing!